src/hotspot/share/prims/jvm.cpp

```diff
@@ -608,7 +608,8 @@ JVM_END
 
 JVM_ENTRY(jint, JVM_IHashCode(JNIEnv* env, jobject handle))
   // as implemented in the classic virtual machine; return 0 if object is null
-  return handle == nullptr ? 0 : ObjectSynchronizer::FastHashCode (THREAD, JNIHandles::resolve_non_null(handle)) ;
+  return handle == nullptr ? 0 :
+         checked_cast<jint>(ObjectSynchronizer::FastHashCode (THREAD, JNIHandles::resolve_non_null(handle)));
```

Comment:
What about making FastHashCode return jint?

Reply:
I agree with Coleen, we use intptr_t all through the markWord code and we can't change all that (not that we would want to) so converting to jint before we leave the VM seems the best place to do it.